### DISCUSSION TITLE
Failed to initialize any reasonable values

### DISCUSSION THREAD

**Discussion by sophie-03** [hahnlab/CAFE5#152] [Q&A] [answered]
I am trying to run CAFE on my output from orthofinder, (54 species and 26341 orthogroups) however I am consistently getting inf likelihoods in my output, which then fails to initialize any reasonable values.

```
Command line: cafe5 -i filtered_cafe_input.txt -t proteome_tree.nwk -c 8 

Filtering families not present at the root from: 23503 to 16396

No root family size distribution specified, using uniform distribution

Optimizer strategy: Nelder-Mead with similarity cutoff
Iterations: 300
Expansion: 2
Reflection: 1

Starting Search for Initial Parameter Values
Lambda: 0.69369833447215
Score (-lnL):             inf
Lambda: 0.45570342302491
Score (-lnL):             inf
Lambda: 0.091953907280693
Score (-lnL):             inf
Lambda: 0.15737163816521
Score (-lnL):             inf
Lambda: 0.080657029957048
Score (-lnL):             inf
Lambda: 0.094100473980456
Score (-lnL):             inf
Lambda: 0.59973101098427
Score (-lnL):             inf
..........................
Lambda: 0.90772462731408
Score (-lnL):             inf
Failed to initialize any reasonable values

```

I have seen other issues with this problem and have tried out the solutions that have been suggested i.e. using the ```clade_and_size_filter.py``` script to filter out gene families with >100 gene copies in one species, which leaves 23503 orthogroups.

I also saw a solution where tree branch lengths needed dividing to make them smaller, however my tree doesn't have any extremely long branches:
[proteome_tree.txt](https://github.com/hahnlab/CAFE5/files/13034576/proteome_tree.txt)

Thanks in advance!

**Comment by benfulton** [comment, marked as answer]
In your case I think your branch lengths may be too small. Most people report it per million years, branch lengths that are in different units will make lambda different. You could try multiplying them instead.

**Reply by sophie-03** [reply]
Thanks for your response. I have tried using a different tree and I am getting the error: 
`Invalid branch length set for Halichoerus_grypusLontra_canadensisMustela_putoriusPhoca_vitulinaZalophus_californianus `
I think this is because there is a 0 branch length. Does cafe deal with branch lengths of 0?

Tree: 
[proteome_tree.nwk.ultrametric.txt](https://github.com/hahnlab/CAFE5/files/13061753/proteome_tree.nwk.ultrametric.txt)

**Reply by benfulton** [reply]
No, branch lengths of 0 aren't allowed. You should give it some small value; but also you should keep your tree ultrametric, so anything you add to that branch should be subtracted from a related branch.

**Comment by sophie-03** [comment]
Thanks for your help. I managed to fix this issue by using a different tree, and CAFE was able to run successfully. I am, however, now having another issue.

I have run CAFE successfully on a set of 16481 orthogroups for 94 species. I now want to subset those orthogroups, so I have a file containing a subset of 294 orthogroups for the same 94 species. I am using the same tree:
[tree.txt](https://github.com/hahnlab/CAFE5/files/13430833/tree.txt).

However, I am getting the same error as previously: inf likelihoods and then "Failed to initialize any reasonable values". Do you have any idea why it fails to initialize when I am using a subset of previous data that did work successfully?
